server.js

```diff
@@ -24,11 +18,123 @@ const app = express();
+  "language_code": String,
+  "num_pages": Number,
+  "ratings_count": Number,
+  "text_reviews_count": Number
+})
+
+
+
+if (process.env.RESET_DB) {
+  const seedDatabase = async () => {
+    await Book.deleteMany({})
+
+    data.forEach((booksData) => {
+      new Book(booksData).save()
+    })
+  }
+
+  seedDatabase()
+}
+
+
+// Home page - first route
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  res.send("Welcome to the website where you can find your favourite book!");
+});
+
+// Route with all the books - full API
+app.get("/books", async (req, res) => {
+  try {
+    const books = await Book.find();
+    res.json(books);
+  } catch (error) {
+    res.status(500).json({ error: "An error occurred while fetching books" });
+  }
+});
+
+
+// Ge books based on number of pages (needs to be befor bookID otherwise that route will override this)
+app.get("/books/pages", async (req, res) => {
+  const { min, max } = req.query;
+
+  // Parse min and max values or assign default values
+  const minPages = parseInt(min, 10) || 0; // Default: 0 pages
+  const maxPages = parseInt(max, 10) || Number.MAX_SAFE_INTEGER; // Default: No upper limit
+
+  console.log(`Filtering books with num_pages between ${minPages} and ${maxPages}`);
+
+  try {
+    // Query to find books within the range
+    const numberOfPages = await Book.find({
+      num_pages: { $gte: minPages, $lte: maxPages },
+    });
+
+    // Return books if found, otherwise send 404
+    if (numberOfPages.length > 0) {
+      res.json(numberOfPages);
+    } else {
+      res.status(404).json({
+        error: `No books found with page count between ${minPages} and ${maxPages}`,
+      });
+    }
+  } catch (error) {
+    // Handle any server-side errors
+    res.status(500).json({ error: "An error occurred while fetching books" });
+  }
+});
+
+
+// Get individual books from id
+app.get("/books/:bookID", async (req, res) => {
+  const bookID = parseInt(req.params.bookID);
+
+  try {
+    const book = await Book.findOne({ bookID: bookID });
+
+    if (book) {
+      res.json(book); // return the specifik book
+    } else {
+      res.status(404).json({ error: "Book not found" }); // error page if book not founf
+    }
+  } catch (error) {
+    res.status(500).json({ error: "An error occurred while fetching the book" }); // 500 page when wrong
+  }
 });
 
+
+// Get book-books by author
+app.get("/books/authors/:author", async (req, res) => {
+  const author = req.params.author;
+
+  try {
+    // $regex = to match name and not case sensitive
+    const booksByAuthor = await Book.find({
+      authors: { $regex: author, $options: "i" }, // "i" case-insensitive
+    });
+
+    if (booksByAuthor.length > 0) {
+      res.json(booksByAuthor); // Returnera böckerna om de finns
+    } else {
+      res.status(404).json({ error: "No books found for this author" });
+    }
+  } catch (error) {
+    res.status(500).json({ error: "An error occurred while fetching books" });
+  }
+});
```

Comment:
And this should also be a query under your /books endpoint to make it more RESTful. Even though you're filtering on author, you still return _books_